pwndbg/commands/context.py

```diff
@@ -180,23 +196,52 @@ def context_code():
         if closest_line < 0:
             return []
 
-        source = gdb.execute('list %i' % closest_line, from_tty=False, to_string=True)
+        # Get the full source code
+        filename = symtab.fullname()
+        source = get_highlight_source(filename)
 
         # If it starts on line 1, it's not really using the
         # correct source code.
         if not source or closest_line <= 1:
             return []
 
-        # highlight the current code line
-        source_lines = source.splitlines()
-        if pwndbg.config.highlight_source:
-            for i in range(len(source_lines)):
-                if source_lines[i].startswith('%s\t' % closest_line):
-                    source_lines[i] = C.highlight(source_lines[i])
-                    break
-
-        banner = [pwndbg.ui.banner("source")]
-        banner.extend(source_lines)
+        n = int(source_code_length)
+
+        # Compute the line range
+        st = max(closest_line - 1 - n//2, 0)
+        ed = min(closest_line - 1 + n//2 + 1, len(source))
```

Comment:
Guess those are `start` and `end` - can we change to those names? `st` and `ed` are not so explicit/known

Reply:
Fixed